src/App.tsx

```diff
@@ -17,18 +53,28 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter onSelect={setStatus} onQueryChange={applyQuery} />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todos.length <= 0 ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={filteredTodos}
+                  setModalOpen={setIsModalOpen}
+                  selectTodo={setSelectedTodo}
+                  isOpen={isModalOpen}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {isModalOpen && (
```

Comment:
```suggestion
      {selectedTodo && (
```
and just check if selectedTodo exists here